pkg/tmplexec/exec.go

```diff
@@ -182,8 +182,10 @@ func (e *TemplateExecuter) Execute(ctx *scan.ScanContext) (bool, error) {
 			// non-`global-matchers` templates can enter the `writeFailureCallback`
 			// func to log failure output.
 			wr := writer.WriteResult(event, e.options.Output, e.options.Progress, e.options.IssuesClient)
-			if wr && !isGlobalMatchers {
-				matched.Store(true)
+			if wr {
+				if !isGlobalMatchers {
+					matched.Store(true)
+				}
```

Comment:
aren’t both of them (this & cc5c5509dc8d87f8b0779b0dfa1e1b9f6f31b5fb) logically equivalent and producing the same result?